packages/web-components/fast-foundation/src/fixture.ts

```diff
@@ -0,0 +1,92 @@
+import {
+    defaultExecutionContext,
+    ExecutionContext,
+    HTMLView,
+    ViewTemplate,
+} from "@microsoft/fast-element";
+
+export interface FixtureOptions {
+    document?: Document;
+    parent?: HTMLElement;
+    source?: any;
+    context?: ExecutionContext;
+}
+
+export interface Fixture<TElement = HTMLElement> {
+    document: Document;
+    template: ViewTemplate;
+    view: HTMLView;
+    parent: HTMLElement;
+    element: TElement;
+    connect(): Promise<void>;
+    disconnect(): Promise<void>;
+}
+
+function findElement(view: HTMLView): HTMLElement {
+    let current: Node | null = view.firstChild;
+
+    while (current !== null && current.nodeType !== 1) {
+        current = current.nextSibling;
+    }
+
+    return current as any;
+}
+
+export function uniqueElementName(): string {
+    return `fast-unique-${Math.random().toString(36).substring(7)}`;
+}
+
+export async function fixture<TElement = HTMLElement>(
+    templateOrElementName: ViewTemplate | string,
+    options: FixtureOptions = {}
+): Promise<Fixture<TElement>> {
+    if (typeof templateOrElementName === "string") {
+        const html = `<${templateOrElementName}></${templateOrElementName}>`;
+        templateOrElementName = new ViewTemplate(html, []);
+    }
+
+    const view = templateOrElementName.create();
+    const document = options.document || globalThis.document;
+    const parent = options.parent || document.createElement("div");
+    const source = options.source || {};
+    const context = options.context || defaultExecutionContext;
+    const element = findElement(view) as any;
+    let isConnected = false;
+
+    view.bind(source, context);
+    view.appendTo(parent);
+
+    customElements.upgrade(parent);
+
+    await Promise.resolve();
```

Comment:
Why are you awaiting empty promises and not storing the value? Is this just to queue a new microtask and give the above time to render to the DOM?

Reply:
Yep! Just building in the micro task await to ensure everything is settled.

Reply:
I also chose to make everything async for consistency sake around this, and to prevent any future breaking changes if additional async work needs to happen within the helpers.

Reply:
Gotcha makes sense. Could you add a comment here addressing that question? I could see this being a bit confusing if folks aren't aware of the inner workings of fast-element or the task queue. Pretty cool though!

Reply:
Definitely!